src/audio/dai-zephyr.c

```diff
@@ -153,25 +154,25 @@ int dai_set_config(struct dai *dai, struct ipc_config_dai *common_config,
 	switch (common_config->type) {
 	case SOF_DAI_INTEL_SSP:
 		cfg.type = is_blob ? DAI_INTEL_SSP_NHLT : DAI_INTEL_SSP;
-		cfg_params = is_blob ? spec_config : &sof_cfg->ssp;
+		cfg_params = is_blob ? (void *)&gtw_cfg->config_data : &sof_cfg->ssp;
```

Comment:
The cast was added to avoid compiler warning like this (being not related to `cfg_params`):
```
pointer type mismatch ('uint32_t const (*)[1]' and 'const struct sof_ipc_dai_ssp_params *') [-Wpointer-type-mismatch]
                cfg_params = is_blob ? &gtw_cfg->config_data : &sof_cfg->ssp;
                                     ^ ~~~~~~~~~~~~~~~~~~~~~   ~~~~~~~~~~~~~
```
Changing for two casts for better readability:
`cfg_params = is_blob ? (void *)&gtw_cfg->config_data : (void *)&sof_cfg->ssp;`